ddl/index.go

```diff
@@ -49,6 +49,10 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.Gen("column does not exist: %s", ic.Column.Name)
 		}
 
+		if col.Flen == 0 {
```

Comment:
It is not a grammar check here. #3838 and #3835 will be fixed in plan/validator.go